taken from this openwrt commit https://dev.openwrt.org/changeset/23905

more information can be found in https://bugs.busybox.net/show_bug.cgi?id=2089

the issue has been fixed for mips/nptl by this upstream commit: http://git.uclibc.org/uClibc/commit/?id=8b48f745be1e086d6e486bbb8167e770f3d1fbc5

--- libpthread/linuxthreads.old/errno.c
+++ libpthread/linuxthreads.old/errno.c
@@ -22,6 +22,7 @@
 #include "internals.h"
 #include <stdio.h>
 
+libpthread_hidden_proto(__errno_location)
 int *
 __errno_location (void)
 {
@@ -29,6 +30,7 @@ __errno_location (void)
   return THREAD_GETMEM (self, p_errnop);
 }
 
+libpthread_hidden_proto(__h_errno_location)
 int *
 __h_errno_location (void)
 {
